### ISSUE TITLE
[Feature] Publish Docker for ARM platform

### ISSUE THREAD

**Issue by glefloch** [dbt-labs/dbt-bigquery#1326]
### Is this your first time submitting a feature request?

- [X] I have read the [expectations for open source contributors](https://docs.getdbt.com/docs/contributing/oss-expectations)
- [X] I have searched the existing issues, and I could not find an existing issue for this feature
- [X] I am requesting a straightforward extension of existing dbt-bigquery functionality, rather than a Big Idea better suited to a discussion

### Describe the feature

The current docker image only support `linux/amd64` platform. It makes it impossible to run the image on ARM

### Describe alternatives you've considered

In order to test on a local Mac, I have to build the image locally.
After looking at the release script, it seems that in input parameters of the docker release script, we can easily specify target platforms  (https://github.com/dbt-labs/dbt-release/blob/8e581cac0fdcf3cefc7a72736d4d71f51420d7b5/.github/workflows/release-docker.yml#L32). 

### Who will this benefit?

People who run M1/M2 macbook would be able to run the image locally. Also, AWS offers Graviton processors based on ARM, people running on those machine could also use it.

### Are you interested in contributing this feature?

yes!

**Comment by amychen1776** [comment]
@glefloch if you feel like contributing this to the adapter, we would very much appreciate that!

**Comment by glefloch** [comment]
Sure, I will try to push something!

**Comment by mikaelahlinder** [comment]
@glefloch Any update on this?

**Comment by glefloch** [comment]
Yes, I have a working version on my machine, but I had to remove a package, and I'm not sure of impacts, I'm running more tests. I will try to open a draft PR ASAP

**Comment by mjdavidson** [comment]
@glefloch how did you get on with this?
